server/graphql/types.js

```diff
@@ -428,8 +434,8 @@ export const ResponseType = new GraphQLObjectType({
       },
       user: {
         type: UserType,
-        resolve(response) {
-          return response.getUser();
+        resolve(response, args, req) {
+          return response.getUserForViewer(req.remoteUser);
```

Comment:
I see what you mean and I don't disagree but I also think that this current solution is good enough for our current use case.

If it was an easy change, I'd be fine with it but turns out it's not trivial because the `email` resolver only has access to the parent `User` and the context (which contains `req.remoteUser`). It doesn't have access to the group. Therefore, we don't know if the remoteUser canEditGroup.